Feature/image tests

ArturGajowy opened the pull request:
This adds a mechanism for defining and running image tests

kokosing commented on teradatalabs/hdp2.5-hive/files/conf-setup/hdfs-site.xml:
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
```
can say something more in the commit message? what was the reason of failing?

kokosing commented on Makefile:
```diff
@@ -145,7 +145,18 @@ $(FLAGDIR)/%.flags: %/Dockerfile $(FLAG_SH)
+	  export EXPECTED_CAPABILITIES=`cat $*/capabilities.txt` && \
+	  cd test && \
+	  docker-compose up -t 0 -d hadoop-master && \
+	  time docker-compose run --rm test-runner
```
[very optional]
I think it should be two separated tasks: all, build and tests, where all could run bulid and tests afterwards

ArturGajowy replied:
My thought exactly, altough implementing it seems like not worth it...

kokosing commented on test/Dockerfile:
```diff
@@ -0,0 +1,27 @@
+  && sed -i '/^baseurl=/c\baseurl=https://archive.cloudera.com/cdh5/redhat/6/x86_64/cdh/5.8.2/' /etc/yum.repos.d/cloudera-cdh5.repo \
+
+  # Install BATS bash test lib
+  && wget https://github.com/ArturGajowy/bats/archive/6959d91.zip -O /tmp/bats.zip \
```
why not to add this archive to this repo and keep as a source file?

kokosing replied:
it looks like bats is not used here yet, so maybe lets remove it for now?

ArturGajowy replied:
> why not to add this archive to this repo and keep as a source file?

Because it ensures the lib is accessible **somewhere**, and makes it clear where :) Otherwise we'd have to put a comment in there, and - as you know - all comments eventually lie

> it looks like bats is not used here yet, so maybe lets remove it for now?

yeah, it should come with the tests commit. Will fix that

kokosing commented on test/Dockerfile:
```diff
@@ -0,0 +1,27 @@
+FROM teradatalabs/centos6-java8-oracle
```
no copyrights?

ArturGajowy commented:
@kokosing comments applied

findepi commented on test/Dockerfile:
```diff
@@ -0,0 +1,50 @@
+    hive \
+
+  # setup CDH repo, pin the CDH distribution to a concrete version
+  && wget http://archive.cloudera.com/cdh5/one-click-install/redhat/6/x86_64/cloudera-cdh-5-0.x86_64.rpm \
```
Shouldn't the "5.0" be extracted to a variable (shell-level or docker-level -- no difference)

findepi commented on test/Dockerfile:
```diff
@@ -0,0 +1,50 @@
+  && yum --nogpgcheck localinstall -y cloudera-cdh-5-0.x86_64.rpm \
+  && rm cloudera-cdh-5-0.x86_64.rpm \
+  && rpm --import http://archive.cloudera.com/cdh5/redhat/6/x86_64/cdh/RPM-GPG-KEY-cloudera \
+  && sed -i '/^baseurl=/c\baseurl=https://archive.cloudera.com/cdh5/redhat/6/x86_64/cdh/5.8.2/' /etc/yum.repos.d/cloudera-cdh5.repo \
```
and, maybe 5.8.2 ?

ArturGajowy replied:
we're not using the values elsewhere, so I'd wait till we do

ArturGajowy replied:
note: if it were a .java file I would gladly apply this comment :)

findepi replied:
Yes, but they are repeated here -- the "5-0" appears thrice. 

Of course, this is minor, can stay as is

ArturGajowy replied:
let's leave it for now, I plan to remove those in favour of local repos anyway

ArturGajowy committed:
Disable EPEL repo in base images

ArturGajowy committed:
Add dockerize to the base images

ArturGajowy committed:
Fix namenode not starting error in hdp2.5-hive

Previously the namenode would be restarted by
supervisor.d because of exiting due to the error

org.apache.hadoop.hdfs.server.common.InconsistentFSStateException: Directory $DIR is in an inconsistent state: storage directory does not exist or is not accessible.

where $DIR was something along the lines of

/tmp/hadoop/dfs/name

This commit fixes it as suggested in
http://stackoverflow.com/questions/6447885/no-namenode-error-in-pseudo-mode

Most likely, the default namenode dir has changed between verisons.

ArturGajowy committed:
Do not fail `pull_latest_images.sh` on missing (new) images

ArturGajowy committed:
Limit `wget` noise by adding `-nv` flag in all usages

ArturGajowy committed:
No longer `pull_latest_images.sh` on Travis

since each commit changes the image labels
and causes a full rebuild of all-but-top-level images.

ArturGajowy committed:
Split travis build into more jobs

for faster buildtimes and more stable build (timeouts!)

ArturGajowy committed:
Add image tests

For now, just the {cdh5,hdp2.5}-hive and images are tested

ArturGajowy committed:
Add `*.dependants` targerts for building along all affected images

ArturGajowy commented:
Merged with #11
